lib/shopify_api/api_version.rb

```diff
@@ -84,5 +66,68 @@ def construct_api_path(_path)
     def construct_graphql_path
       raise NotImplementedError
     end
+
+    protected
+
+    attr_reader :numeric_version
+
+    class NoVersion < ApiVersion
+      API_PREFIX = '/admin/'
+
+      def initialize
+        @version_name = "no_version"
+        @numeric_version = 0
+      end
+
+      def construct_api_path(path)
+        "#{API_PREFIX}#{path}"
+      end
+
+      def construct_graphql_path
+        '/admin/api/graphql.json'
+      end
+    end
+
+    class Unstable < ApiVersion
+      API_PREFIX = '/admin/api/unstable/'
+
+      def initialize
+        @version_name = "unstable"
+        @url = API_PREFIX
+        @numeric_version = 9_000_00
+      end
+
+      def construct_api_path(path)
+        "#{@url}#{path}"
+      end
+
+      def construct_graphql_path
+        construct_api_path("graphql.json")
+      end
+    end
+
+    class Release < ApiVersion
+      FORMAT = /^\d{4}-\d{2}$/.freeze
+      API_PREFIX = '/admin/api/'
+
+      def initialize(version_number)
+        raise InvalidVersion, version_number unless version_number.match(FORMAT)
+        @version_name = version_number
+        @url = "#{API_PREFIX}#{version_number}/"
+        @numeric_version = version_number.tr('-','').ljust(6, '0').to_i
```

Comment:
legit warning

Reply:
Also re: `version_number.tr('-','').ljust(6, '0').to_i` I think you said this would still function as expected if version handles included days of the month, i.e. `2019-03-15`. I think:
* a) it would be better to avoid jumping levels of abstraction (from version names, urls, to cryptic primitive string, integer manipulation) and put this logic into a well-named private method. and
* b) add a test to verify the `ljust` works as expected, perhaps asserting `2019-03-15` > `2019-03` > `2019-02-15`